spanner/integration_test.go

```diff
@@ -489,16 +513,39 @@ func TestIntegration_SingleUse(t *testing.T) {
 		},
 	} {
 		t.Run(test.name, func(t *testing.T) {
-			// SingleUse.Query
+			var got [][]interface{}
 			su := client.Single().WithTimestampBound(test.tb)
-			got, err := readAll(su.Query(
-				ctx,
-				Statement{
-					"SELECT SingerId, FirstName, LastName FROM Singers WHERE SingerId IN (@id1, @id3, @id4) ORDER BY SingerId",
-					map[string]interface{}{"id1": int64(1), "id3": int64(3), "id4": int64(4)},
-				}))
-			if err != nil {
-				t.Fatalf("%d: SingleUse.Query returns error %v, want nil", i, err)
+			// SingleUse.Query
+			if testDialect == adminpb.DatabaseDialect_POSTGRESQL {
+				if test.skipForPG {
+					t.Skip("Skipping testing of unsupported tests in Postgres dialect.")
+				}
+				// In clause not supported in PG dialect b/186271659
+				for _, id := range []int64{1, 3, 4} {
```

Comment:
Wouldn't it in that case be better to re-write the query to something like `WHERE SingerId=@id1 OR SingerId=@id3 OR SingerId=@id4` instead of this loop. You could probably also use the same query for both PG and Spanner in that case, which would mean that this entire block could probably be simplified even more.

Reply:
Thanks @olavloite I have refactored the test to use query as per your suggestion and its passing, but we can't use same query for both PG and Spanner since query parameter parsing logic is different.

Reply:
Ah, yeah, I forgot that the query parameters in PG look different. But this looks good to me.